gammapy/background/tests/test_energy_offset_array.py

```diff
@@ -0,0 +1,28 @@
+from ..energy_offset_array import EnergyOffsetArray
+from gammapy.data import DataStore
+from gammapy.data import EventList
+from ...datasets import gammapy_extra
+from ...utils.testing import requires_dependency, requires_data
+import numpy as np
+#import pylab as pt
+
+
+@requires_dependency('matplotlib')
+@requires_data('gammapy-extra')
+def test_energy_offset_array():
+    dir=str(gammapy_extra.dir)+'/datasets/hess-crab4'
+    data_store = DataStore.from_dir(dir)
+    Observation_Table=data_store.obs_table
+    event_list_files = data_store.make_table_of_files(Observation_Table,
+                                                          filetypes=['events'])
+    ev_list=[]
+    for (i,i_ev_file) in enumerate(event_list_files['filename']):
+        ev_list.append(EventList.read(i_ev_file))
+        energy=np.logspace(-1,2 , 100)
+        offset=np.linspace(0, 2.5, 100)
+        Array=EnergyOffsetArray(energy,offset)
+        Array.fill_events(ev_list)
+        g=Array.plot_image()
+        g.figure.savefig("/Users/jouvin/gammapy/gammapy/background/tests/test_energy_offset_array.png")
```

Comment:
ok!
I took your other comments into account! tell me if this is better! the code looks better(-:
Le 20 janv. 2016 à 14:07, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/background/tests/test_energy_offset_array.py:
> 
> > +@requires_data('gammapy-extra')
> > +def test_energy_offset_array():
> > +    dir=str(gammapy_extra.dir)+'/datasets/hess-crab4'
> > +    data_store = DataStore.from_dir(dir)
> > +    Observation_Table=data_store.obs_table
> > +    event_list_files = data_store.make_table_of_files(Observation_Table,
> > +                                                          filetypes=['events'])
> > +    ev_list=[]
> > +    for (i,i_ev_file) in enumerate(event_list_files['filename']):
> > +        ev_list.append(EventList.read(i_ev_file))
> > +        energy=np.logspace(-1,2 , 100)
> > +        offset=np.linspace(0, 2.5, 100)
> > +        Array=EnergyOffsetArray(energy,offset)
> > +        Array.fill_events(ev_list)
> > +        g=Array.plot_image()
> > +        g.figure.savefig("/Users/jouvin/gammapy/gammapy/background/tests/test_energy_offset_array.png")
> My suggestion would be that you make a new page docs/background/energy_offset_array.rst.
> We might group things better in the future, but for now, a new page is simplest.
> Start by copy & pasting one of the existing pages, and add energy_offset_array to the toctree directive in docs/background/index.rst so that it shows up in the docs build.
> 
> OK?
> 
> —
> Reply to this email directly or view it on GitHub.
>